src/Microsoft.Dnx.Runtime/Compilation/ICompilationSession.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.Dnx.Runtime.Compilation
+{
+    public interface ICompilationSession : IDisposable
+    {
+        event Action<string> OnInputFileChanged;
+
+        Assembly CompileAndLoadProject(Project project, CompilationTarget target, IAssemblyLoadContext loadContext);
```

Comment:
boxes and lines were drawn, I'll adjust the name to `LoadProject` I think, otherwise we're good here.